### ISSUE TITLE
Youtube Link Title no longer works/installable

### ISSUE THREAD

**Issue by Lunarhawk** [awesome-scripts/awesome-userscripts#27]
I recently moved from Waterfox Classic back to Firefox and discovered YLT no longer seems to work at all in FF.  I've tried a few things to fix it but it seems like its no longer even installable (throwing a 403 error from its original host?) or updateable, and even if you use an offshoot fork someone made (azzizLight) to get around that it seems like it still doesn't work at all anymore in modern FF

Just letting you know that you could prob remove it and also ask if you know if anything that does exactly the same thing in terms of previewing YT vids from links/replacing the urls with their Titles that still works now in 2023?

**Comment by adamlui** [comment]
@Lunarhawk I confirmed YLT doesn't work in Chrome either, but just tested this script does the hover previews from links: https://greasyfork.org/en/scripts/370755-youtube-peek-preview

**Comment by Lunarhawk** [comment]
That doesnt work in FF either which makes me wonder if its an FF greasemonkey thing now. Which would be a shame if that's the case

EDIT: Confirmed its a greasemonkey thing, swapping to Tampermonkey seems to make them work. That's a damn shame guess gm just kinda sucks now? idk

**Comment by adamlui** [comment]
@Lunarhawk actually I used FF to test it working:

![image](https://user-images.githubusercontent.com/10906554/229652050-3ffd740d-ac18-45ea-9acf-5a8fdc99ffbf.png)


That appears when I hover over the link "Full-Window mode" (which previously was the alt-tag of an image thumbnail linking to YouTube) of this page: https://github.com/adamlui/chatgpt-widescreen

Can you try if it works on that page (then I can figure out if it's the script or your browser)

**Comment by adamlui** [comment]
> That doesnt work in FF either which makes me wonder if its an FF greasemonkey thing now. Which would be a shame if that's the case
> 
> EDIT: swapping to Tampermonkey seems to make them work. That's a damn shame guess gm just kinda sucks now? idk

Ohhhhhhh I didn't see you say greasemonkey! Yeah that userscript manager is very incompatible with many scripts, for some reason the developer decided to outright not load scripts that use the `GM_` methods (probably killing hundreds of thousands of userscsripts)

**Comment by qsniyg** [comment]
@adamlui I believe it was due to the switch from XUL to WebExtensions. Though it's still possible to implement most of the the `GM_` APIs through WebExtensions (with a few exceptions, such as the `synchronous` flag for `GM_xmlHttpRequest`).

**Comment by Lunarhawk** [comment]
Yeah I've been on Waterfox Classic for yearssss for reasons that ended yesterday so I was just out of the loop on GM being bad now. Thanks for taking the time to test it out so quickly tho!

**Comment by adamlui** [comment]
Curious what were the reasons? because I actually downloaded the installer but never got around to using it

**Comment by Lunarhawk** [comment]
It's nothing that would be useful for you, there was a single site I used daily that was made 10x nicer with a comprehensive extension for it that the person stopped updating before FF's big change that killed nearly every addon if they weren't updated so I swapped to WFC specifically to keep using the site that way. And the site shut down yesterday so there went my reason to keep using it lmao

**Comment by garoto** [comment]
Hey, more than three years later, and I I' writing this to say that the latest version of "YouTube Link Title" released by kuehlschrank in 2017.2.28 works on modern Firefoxes fine. What stopped working some 5yrs ago or so was the YT apikey that was embedded in the script. Replacing the outdated key with your own makes it work again no problem.
